src/app/routes/enrolment/utils/format-validity-period-string.ts

```diff
@@ -0,0 +1,60 @@
+export const formatValidityPeriod = (num: number): string => {
+  if (!num) {
+    return null;
+  }
+
+  enum UnitsOfTime {
+    YEAR = 'year',
+    YEARS = 'years',
+    DAY = 'day',
+    DAYS = 'days',
+    HOUR = 'hour',
+    HOURS = 'hours',
+    MINUTE = 'minute',
+    MINUTES = 'minutes',
+    SECOND = 'second',
+    SECONDS = 'seconds',
+  }
+
+  const years = {
```

Comment:
I'm wondering if a library like momentJS already does calculations like this...

Reply:
I think momentJS is not. And i would not use it any more, because it is deprecated and it would be better to remove it and use something new. Anyway there is a pipe `timeDuration` that is handling this -> it's already tested, you can check if it fits your needs.

Reply:
I think that is taking the individual values rather than parsing from a timestamp? should I move this logic to that class?

Reply:
It's parsing from seconds to readable years/days/hours/minutes/seconds text (check tests). No need to move this logic, because that pipe is already doing the same thing. Look for example for role creation where you're setting defaultValidityPeriod.